inference/generativeai/llm-workshop/lab4-openchatkit/deploy_openchatkit_on_sagemaker_deepspeed.ipynb

```diff
@@ -0,0 +1,1141 @@
+    "        },\n",
+    "    ],\n",
+    ")\n",
+    "endpoint_config_response"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "53d84ba6",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "create_endpoint_response = sm_client.create_endpoint(\n",
+    "    EndpointName=f\"{endpoint_name}\", EndpointConfigName=endpoint_config_name\n",
+    ")\n",
+    "print(f\"Created Endpoint: {create_endpoint_response['EndpointArn']}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d7454013",
+   "metadata": {},
+   "source": [
+    "### This step can take ~ 10 min or longer so please be patient"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0de9b6eb",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import time\n",
+    "\n",
+    "resp = sm_client.describe_endpoint(EndpointName=endpoint_name)\n",
+    "status = resp[\"EndpointStatus\"]\n",
+    "print(\"Status: \" + status)\n",
+    "\n",
+    "while status == \"Creating\":\n",
+    "    time.sleep(60)\n",
+    "    resp = sm_client.describe_endpoint(EndpointName=endpoint_name)\n",
+    "    status = resp[\"EndpointStatus\"]\n",
+    "    print(\"Status: \" + status)\n",
+    "\n",
+    "print(\"Arn: \" + resp[\"EndpointArn\"])\n",
+    "print(\"Status: \" + status)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "932d8421",
+   "metadata": {},
+   "source": [
+    "#### While you wait for the endpoint to be created, you can read more about:\n",
+    "- [Deep Learning containers for large model inference](https://docs.aws.amazon.com/sagemaker/latest/dg/realtime-endpoints-large-model-dlc.html)\n",
+    "- [DeepSpeed](https://www.deepspeed.ai/tutorials/inference-tutorial/#initializing-for-inference)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "12f1fa1b",
+   "metadata": {},
+   "source": [
+    "#### Leverage the Boto3 to invoke the endpoint. \n",
+    "\n",
+    "This is a generative model so we pass in a Text as a prompt and Model will complete the sentence and return the results.\n",
+    "\n",
+    "You can pass a batch of prompts as input to the model. This done by setting `inputs` to the list of prompts. The model then returns a result for each prompt. The text generation can be configured using appropriate parameters. These `parameters` need to be passed to the endpoint as a dictionary of `kwargs`. Refer this documentation - https://huggingface.co/docs/transformers/main/en/main_classes/text_generation#transformers.GenerationConfig for more details.\n",
+    "\n",
+    "The below code sample illustrates the invocation of the endpoint using a prompts and also sets some parameters.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7c9ff1d6",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "prompts = [\n",
```

Comment:
could we wrap this part as a conversation function? It could be as simple as something like below:

```
def chat(prompt):
     prompt = previous_chain_prompts + "\n<human>: " + prompt
     previous_chain_prompts = predictor.predict(prompt)
     return previous_chain_prompts[len(prompt):]

chat(Hello)
>>> Hi, how are you doing today
```
Customer could run as much as they want

Reply:
Great suggestion! We should have this once we enhance this notebook to include conversational capabilities including information retrieval integration with OpenChatKit. We will incorporate this in our next iteration of this notebook. For now, I would like to keep it basic.